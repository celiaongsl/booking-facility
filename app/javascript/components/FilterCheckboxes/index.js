import React from "react";
import {
  FormControl,
  FormGroup,
  Typography,
  FormControlLabel,
  Checkbox,
} from "@material-ui/core";

import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
  },
  formControl: {
    margin: theme.spacing(3),
  },
}));

const FilterCheckboxes = (props) => {
  const { title = "", checkboxList = {}, setCheckboxList = () => null } = props;

  const classes = useStyles();

  return (
    <React.Fragment>
      <Typography variant="h6">{title}</Typography>
      <FormControl component="fieldset" className={classes.formControl}>
        <FormGroup data-testid="checkbox-list">
          {Object.keys(checkboxList).map((key, index) => {
            return (
              <FormControlLabel
                key={index}
                control={
                  <Checkbox
                    // data-testid="test-checkbox"
                    // data-testid={index}
                    checked={checkboxList[key]}
                    onChange={(event) =>
                      setCheckboxList({
                        ...checkboxList,
                        [event.target.name]: event.target.checked,
                      })
                    }
                    name={key}
                    color="primary"
                  />
                }
                label={title === "Floor" ? `Floor ${key}` : `${key} pax`}
              />
            );
          })}
        </FormGroup>
      </FormControl>
    </React.Fragment>
  );
};

export default FilterCheckboxes;
