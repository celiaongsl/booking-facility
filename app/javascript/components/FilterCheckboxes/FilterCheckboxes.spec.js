import React from "react";
import FilterCheckboxes from "./";
import { render, fireEvent } from "@testing-library/react";

describe("<FilterCheckboxes />", () => {
  const setup = () => {
    const utils = render(<FilterCheckboxes />);
    // const datepickerInput = utils.getByLabelText("change date");
    return {
      //   datepickerInput,
      ...utils,
    };
  };

  it("renders the component", () => {
    const { asFragment } = render(<FilterCheckboxes />);
    expect(asFragment()).toMatchSnapshot();
  });

});
