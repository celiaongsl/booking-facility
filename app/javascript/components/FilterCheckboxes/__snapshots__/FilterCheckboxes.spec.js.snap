// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterCheckboxes /> renders the component 1`] = `
<DocumentFragment>
  <h6
    class="MuiTypography-root MuiTypography-h6"
  />
  <fieldset
    class="MuiFormControl-root makeStyles-formControl-2"
  >
    <div
      class="MuiFormGroup-root"
    />
  </fieldset>
</DocumentFragment>
`;
